src/WebJobs.Script/Diagnostics/FunctionFileLoggerProvider.cs

```diff
@@ -50,7 +50,7 @@ public ILogger CreateLogger(string categoryName)
             {
                 // Make sure that we return the same fileWriter if multiple loggers write to the same path. This happens
                 // with Function logs as Function.{FunctionName} and Function.{FunctionName}.User both go to the same file.
-                IFileWriter fileWriter = _fileWriterCache.GetOrAdd(filePath, (p) => _fileWriterFactory.Create(Path.Combine(_roogLogPath, filePath)));
+                IFileWriter fileWriter = _fileWriterCache.GetOrAdd(filePath, (path) => _fileWriterFactory.Create(Path.Combine(_roogLogPath, path)));
```

Comment:
You could pass in the fileWriterFactory as an argument by using

https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getoradd?view=net-5.0#System_Collections_Concurrent_ConcurrentDictionary_2_GetOrAdd__1__0_System_Func__0___0__1____0_

and then make this static too if I'm not mistaken

Reply:
Oh damn I have exactly this but it didn't make it into the diff branch (maybe be on the other laptop) - thanks for pointing out! I'll get this one amended.

Reply:
Ah I remember where this was now - we need 2 bits of context. I was going to benchmark the ValueTuple needed here vs. the lambda creation per call and see what the overall overhead is...and then forgot. I'll do that Monday and report back :)

Reply:
Ah yeah it needs two contexts. These things are sometimes hard to spot in the diff view only. I confess I didn't checkout the code 🙄

Reply:
Reminder that CD is not locking on `GetOrAdd` so multiple threads can concurrently invoke the lambda, even though a single value will be added to the dictionary. It might be recommended to use the `Lazy` pattern.

Reply:
And I checked `FileWriter` which should be disposed, so it's even more important to not lose these unnecessary instances. I don't know the cardinality of these writers, but they each create a `Timer`, which is expected to be disposed. There is no finalizer, so it might be necessary to add one, even more that the Dispose is calling GC.SuppressFinlaize so it apparently expect one to be here.